docs/examples.rst

```diff
@@ -362,3 +362,77 @@ Calculate the token count from the token balance of an account address.
+
+.. doctest::
+
+    >>> factory=web3.eth.contract(abi=json.loads(abi), bytecode=bytecode)
+    >>> tx_hash = factory.constructor(1000000, 'DEM', 18, 'DE', 1).transact()
+    >>> print(tx_hash)
+    b'\xe9K\xcfz\xfc9*-+\x17|\xd4at.\xe32t\x13\xf4\\\xe8\x8b\x9e6?\x08\xa1\xb4\xe6[\xd3'
+    >>> txn_receipt = web3.eth.waitForTransactionReceipt(tx_hash)
+
+
+Instantiate a contract pointing at a specific address
+-----------------------------------------------------
+
+.. doctest::
+
+    >>> new_contract = web3.eth.contract(address='0x89205A3A3b2A69De6Dbf7f01ED13B2108B2c43e7', abi=json.loads(abi))
+    >>> balance = new_contract.functions.balanceOf('0x89205A3A3b2A69De6Dbf7f01ED13B2108B2c43e7')
+
+Query account balance
+---------------------
+
+.. doctest::
+
+    >>> new_contract = web3.eth.contract(address=web3.eth.accounts[0], abi=json.loads(abi))
+    >>> balance = web3.eth.getBalance(web3.eth.accounts[0])
+
+Perform direct transfer call
+----------------------------
+
+.. doctest::
+
+    >>> new_contract = web3.eth.contract(address=web3.eth.accounts[0], abi=json.loads(abi))
+    >>> txn_hash = new_contract.functions.transfer(web3.eth.accounts[1], 12345).transact()
+    >>> txn_receipt = web3.eth.waitForTransactionReceipt(txn_hash)
+
+Querying the approved spending balance for external transfers
+-------------------------------------------------------------
+
+.. doctest::
+
+    >>> new_contract = web3.eth.contract(address=web3.eth.accounts[0], abi=json.loads(abi))
+    >>> txn_hash = new_contract.functions.transferFrom(web3.eth.accounts[0], web3.eth.accounts[1], 12345).transact()
+    >>> txn_receipt = web3.eth.waitForTransactionReceipt(txn_hash)
+
+Creating an approval for external transfers
+-------------------------------------------
+
+.. doctest::
+
+    >>> new_contract = web3.eth.contract(address=web3.eth.accounts[0], abi=json.loads(abi))
+    >>> txn_hash = new_contract.functions.approve(web3.eth.accounts[0], 12345).transact()
+    >>> txn_receipt = web3.eth.waitForTransactionReceipt(txn_hash)
+
+Performing a transferFrom call with an approved transfer.
+---------------------------------------------------------
+
+.. doctest::
+
+    >>> new_contract = web3.eth.contract(address=web3.eth.accounts[0], abi=json.loads(abi))
```

Comment:
Maybe instead of using `web3.eth.accounts[0]` we could do something like this.

```python
alice, bob = web3.eth.accounts[:2]
new_contract.functions.transferFrom(alice, bob, 1234)
```

I think it reads a lot cleaner.